lib/masamune/transform/deduplicate_dimension.rb

```diff
@@ -35,8 +35,26 @@ def insert_columns(source = nil)
         consolidated_columns.map { |_, column| column.name }
       end
 
-      def insert_view_values
-        consolidated_columns.map { |_, column| column.name }
+      def insert_view_values(coalesce: false)
+        consolidated_columns.map do |_, column|
+          if !column.default.nil? && coalesce
+            "COALESCE(#{column.name}, #{column.sql_value(column.default)}) AS #{column.name}"
+          else
+            column.name
+          end
+        end
+      end
+
+      def duplicate_value_conditions(window)
+        [].tap do |result|
+          consolidated_columns.map do |_, column|
+            if column.null
+              result << "((LAG(#{column.name}) OVER #{window} = #{column.name}) OR (LAG(#{column.name}) OVER #{window} IS NULL AND #{column.name} IS NULL))"
```

Comment:
this window function to detect duplicate was not working with null columns since `NULL = NULL` is always `FALSE`